docs/integrations/webhooks/debug-your-webhooks.mdx

```diff
@@ -0,0 +1,112 @@
+
+## Webhooks and local development
+
+When you or a user of your application performs certain actions, a webhook can be triggered. You can see the full list of [webhook events](/docs/integrations/webhooks/overview#supported-webhook-events) for a list of the actions that could result in a Webhook. Depending on the events subscribed to in the [Webhook section](https://dashboard.clerk.com/last-active?path=webhooks) of your dashboard, a webhook event will be triggered and sent to the specified endpoint in your application.
+
+When you are developing on your localhost, your application is not internet facing and can't receive the webhook request. You will need to use a tool that creates a tunnel from the internet to your localhost. These tools will provide temporary or permanant URLs depending on the tool and the plan you subscribe to. Popular tools include `ngrok`, `localtunnel` and `Cloudflare Tunnel`.
+
+Below is an image demonstrating the flow from when you perform an action that triggers a webhook through until the webhook is received by your application. This flow chart will show the role of the tunnel.
+
+FLOW CHART IMAGE HERE
+
+With the Webhook flow in mind, the following areas could cause problems when trying to develop with webhooks:
+
+1. Middleware is not configured correctly for webhooks
+2. The webhook route is not correctly configured
+3. Configuration in your dashboard
+4. The tunnel configuration
+5. Verification of the webhook
+6. The logic you created to process the webhook
+
+
+##  Check your Middleware configuration
+
+Incoming webhook events will never be signed in -- they are coming from a source outside of your application. Since they will be in a signed out state, the simplest configuration is to use `ignoredRoutes` for your webhook routes. Below is an example of making all webhook routes ignored.
+
+```tsx filename="middleware.tsx" {4}
+import { authMiddleware } from "@clerk/nextjs";
+
+export default authMiddleware({
+  ignoredRoutes: ["/api/webhooks(.*)"],
+});
+
+export const config = {
+  matcher: ["/((?!.+\\.[\\w]+$|_next).*)", "/", "/(api|trpc)(.*)"],
+};
+```
+
+## Test the Route Handler or API Route
+
+If you are having trouble with your webhook, you can create a basic Route Handler and then test it locally.  First, add the following file and code to your app.
+
+```ts filename="app/webhooks/test/route.ts"
+export async function POST() {
+   return Response.json({ message: 'The route is working' });
+}
+```
+
+While your application is running, try this from the command line.
+
+```bash
+curl -H 'Content-Type: application/json' \
+     -X POST http://localhost:3000/api/webhooks/test
+```
+
+If you see the `{"message":"The route is working"}` then the basic Route Handler is working, and ready to build on.
+
+
+## Check your dashboard configuration
```

Comment:
If we aren't including mentioning the "Clerk Dashboard" explicitly anywhere in this section, then we should at least put it in the title so users know we are talking about the Clerk Dashboard

```suggestion
## Check the webhook configuration in the Clerk Dashboard
```